### PR TITLE
Mask.to_array returns (mask, bounding_box)

### PR DESCRIPTION
**Pull request by drothlis**
This is much nicer for the caller and we don't need `_validate_region`. Thanks to @wmanley for the [suggestion](https://github.com/stb-tester/stb-tester/pull/766#discussion_r914796857).

### CODE REVIEW

File: _stbt/diff.py [hunk 1]
```diff
@@ -35,12 +36,9 @@ def __init__(self, initial_frame, mask=Region.ALL, min_size=None,
         self.prev_frame = initial_frame
         self.min_size = min_size
 
-        mask, region, frame_region = _validate_mask(initial_frame, mask)
-        if mask.is_region():
-            self.mask_ = None
-        else:
-            self.mask_ = crop(mask.to_array(frame_region), region)
-        self.region = region
+        self.mask_, self.region = load_mask(mask).to_array(initial_frame.region)
```
Comment (wmanley) [inline _stbt/diff.py:39]:
```suggestion
        self.mask_, self.region = load_mask(mask).to_array(_image_region(initial_frame))
```

File: _stbt/mask.py [hunk 2]
```diff
@@ -118,21 +120,12 @@ def __hash__(self):
...
-
-        If you're implementing your own image-processing algorithm: With a
-        simple mask you don't need to apply pixel-wise masking, just a bounding
-        box.
-        """
-        return self._region is not None and not self._invert
-
     def to_array(self, region: Region, color_channels: int = 1) \
-            -> numpy.ndarray:
+            -> Tuple[numpy.ndarray, Region]:
```
Comment (wmanley) [inline _stbt/mask.py:124]:
```suggestion
            -> Tuple[Optional[numpy.ndarray], Region]:
```

File: _stbt/mask.py [hunk 3]
```diff
@@ -142,35 +135,31 @@ def to_array(self, region: Region, color_channels: int = 1) \
...
-          are white (255) and masked-out pixels are black (0).
-        """
-        if region.x != 0 or region.y != 0:
-            raise ValueError(
-                f"{region} must be a full-frame region starting at x=0, y=0")
-        return _to_array_cached(self, region, color_channels)
-
-    def bounding_box(self, region: Region) -> Region:
-        """Calculate a bounding box around the masked-in area.
+        :rtype: Tuple[numpy.ndarray, Region]
```
Comment (wmanley) [inline _stbt/mask.py:138]:
```suggestion
        :rtype: Tuple[Optional[numpy.ndarray], Region]
```

**Review by wmanley — COMMENTED** [review]
Nice stuff

File: _stbt/black.py [hunk 4]
```diff
@@ -66,14 +65,18 @@ def is_screen_black(frame: Optional[Frame] = None,
         from stbt_core import get_frame
         frame = get_frame()
 
-    mask, region, frame_region = _validate_mask(frame, mask, region)
+    if region is not Region.ALL:
+        if mask is not Region.ALL:
+            raise ValueError("Cannot specify mask and region at the same time")
+        mask = region
+
+    mask_, region = load_mask(mask).to_array(frame.region)
```
[the remark this answers is not in the document]
Comment (drothlis) [reply _stbt/black.py:73]: Fixed in #774.

File: _stbt/mask.py (hunk 3 above)
Comment (drothlis) [reply _stbt/mask.py:138]: Fixed in #772.

File: _stbt/mask.py (hunk 2 above)
Comment (drothlis) [reply _stbt/mask.py:124]: Fixed in #772.

**Pull request merged**